bika/lims/browser/bika_listing.py

```diff
@@ -1042,15 +1042,11 @@ def get_sort_order(self):
         """Get the sort_order criteria from the request or view
         """
         form_id = self.get_form_id()
-        key = "{}_sort_order".format(form_id)
-        sort_order = self.request.get(key, None)
-
-        if sort_order is None:
-            sort_order = self.contentFilter.get("sort_order", self.sort_order)
-        if sort_order not in ["ascending", "descending"]:
-            sort_order = "descending"
-
-        return sort_order
+        allowed = ["ascending", "descending"]
+        sort_order = [self.request.get("{}_sort_order".format(form_id), None),
+                      self.contentFilter.get("sort_order", None)]
+        sort_order = filter(lambda order: order in allowed, sort_order)
+        return sort_order and sort_order[0] or "descending"
```

Comment:
Puh, these 5 lines took me now 5 minutes to parse... Better keep it simpler, because you never know if it is you who will need to work on this code again 😅

Reply:
Well, you asked me to merge those two conditionals and I ended up with this.. Apologies! BTW, you accepted but did not merge this PR...